--- clean/xclannad/window/picture.cc	2005-04-06 06:04:05.000000000 +0100
+++ shish/xclannad/window/picture.cc	2005-04-14 03:37:32.121809856 +0100
@@ -944,7 +944,12 @@
 	return s->format->Amask != 0;
 }
 void PicRoot::SetWindowCaption(const char* caption) {
-	SDL_WM_SetCaption(caption, 0);
+	if(strlen(caption) == 0) {
+		SDL_WM_SetCaption("XClannad - No Game Loaded", 0);
+	}
+	else {
+		SDL_WM_SetCaption(caption, 0);
+	}
 }
 
 /************************************************************
